File: codeforces/round642/C.py

squares = {
    1:0,
    3:8
}

maxN = 1
T = int(input())
for t in range(T):
    n = int(input())
    if n > maxN:
        for i in range(maxN + 2, n + 2, 2):
            outer = n + (2 * (n - 1)) + (n - 2);
            level = n / 2;
            layer = outer * level;
        maxN = n

File: codeforces/round642/C.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    int T;
    cin >> T;

    map<unsigned long long, unsigned long long> squares;
    squares.insert(make_pair(1, 0));
    squares.insert(make_pair(3, 8));

    unsigned long long maxN = 1;
    for (int t = 0; t < T; t++) {
        unsigned long long n;
        cin >> n;
        if (n > maxN) {
            for (int i = maxN + 2; i <= n; i += 2) {
                unsigned long long outer = i + (2 * (i - 1)) + (i - 2);
                unsigned long long level = i / 2;
                unsigned long long layer = outer * level;
                squares.insert(make_pair(i, layer + squares.at(i - 2)));
            }
            maxN = n;
        }

        unsigned long long total = squares.at(n);
        cout << total << endl;
    }
}
